src/controllers/twoFactorAuth.controller.ts

```diff
@@ -0,0 +1,68 @@
+import { Request, Response } from "express";
+import User from "../database/models/User.model";
+import * as twofactor from "node-2fa";
+import sendEmailToken from "../services/mails/sendEmailToken";
+import { Role } from "../interfaces";
+
+/*
+  this middleware will enable two factor authentication on user account, create random code and send it to user through email and have to be verified to allow user access protected routes 
+*/
+export const create2FAToken = async (req: Request, res: Response) => {
+  try {
+    const newSecret = twofactor.generateSecret({
+      name: "Ecom-backend",
+      account: req.user.surname,
+    });
+
+    const user = await User.findByPk(req.user.id);
+
+    if (user && user?.role !== Role.Seller) {
```

Comment:
I think here you may `import { isSeller} from '../middleware/auth`'; it can do similar function

Reply:
Thanks, this saved me from hound bro